### PR TITLE
Add covariance to estimation msgs

### PR DESCRIPTION
**Pull request by TSC21**
Given some position estimation/filters maybe be feeding data into onboard position and attitude estimators, and since IEKF, LPE, EFK2 may use covariance matrices to improve the estimation, this PR allows sending these same covariances through specific messages (vision, mocap).

@jgoppert this may interest you as one can send the odometry data from external sources through these specific msgs and receive feedback though `LOCAL_POSITION_NED_COV` and `ATTITUDE_QUATERNION_COV` respectively.

### CODE REVIEW

**Comment by jgoppert** [comment]
Yeah, looks good, would be nice to have this in.

**Comment by TSC21** [comment]
Is there anything else to be considered/modified so this can be merged?

**Comment by jgoppert** [comment]
I haven't tested extensions out enough to approve this myself. Do you have any test showing this in use? That would probably help.

**Comment by TSC21** [comment]
Have not tried myself but for what I know extensions exist so they can also be used in MAVLink 2.0. We can probably add a modification to the odom plugin in mavros and check its use, since it will be one of the main usages of this extensions (while a odom msg is not added to the msg API).

**Comment by TSC21** [comment]
@LorenzMeier is there something that can be done to avoid this being staled forever? Thank you

**Comment by LorenzMeier** [comment]
@TSC21 I'm sure I commented on this, but my original post might not have made it through: The issue is that MAVLink 2 is also used on radio links and your changes more than double the required bandwidth. This has most likely detrimental effects on a lot of users. Since MAVLink 2 sends zero-filled packets with reduced length we actually might get away with it but it is a complex topic. I think the better approach is to get this data over our new FastRTPS bridge to ROS instead.

**Comment by TSC21** [comment]
Dear @LorenzMeier,

Thanks for your feedback.

Some matters about your comment:

1. I'm not seeing why people will be sending vision estimates through a radio stream. I can see it happen, for example, to the mocap cases, but that will be in cases where the tests are made in controlled environments and with the FCU radio and the GCS radio close to each other in the same room. For both cases: in vision I think the most common case is to send this same message from a companion board through a serial interface (USB<->UART), as you are probably aware. For mocap cases, I think there are many using wifi modules to do it (like the esp8266 interface that Pixracer uses), where the bandwidth and throughput can be much higher.

2. In any case, this PR can be changed so it doesn't get that heavy in terms of bandwidth, 'cause since the covariance matrices are symmetrical, I can change from a full covariance matrix to only use the URT (that means that a 6x6 matrix will become an array of 19 floats - instead of 36 - and a probably we can keep the 3x3, since it's only 9 floats). Would you accept the PR with that modification? Does it work for you?

3. Are you talking about https://github.com/eProsima/Firmware? I thought this wasn't mainstream already. But this is something to be consider only for ROS2, right? Regardless, this is still a matter that deserves a close follow up.

**Comment by TSC21** [comment]
@julianoes can I get your input here? Thanks

**Comment by julianoes** [comment]
I don't think I have an opinion. @mrivi can you check please?

**Comment by TSC21** [comment]
@mrivi friendly ping

**Comment by mrivi** [comment]
@TSC21 Sorry, I completely missed this. It looks good to me.

**Comment by TSC21** [comment]
@mrivi do you think the matrices sizes shouls be reduces to fit the URT of the matrix instead of the full matrix?

**Comment by mrivi** [comment]
@TSC21 yes, I think we should exploit the fact that the matrices are symmetrical and reduce the bandwidth

**Comment by TSC21** [comment]
Yeah I'll do that. I'm currently travelling but as soon as I can, I will tackle this.

**Comment by TSC21** [comment]
@LorenzMeier, @mrivi I updated the matrices so they use only the URT, given the fact the matrices are symmetrical.

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -3615,12 +3617,16 @@
       <field type="float" name="roll" units="rad">Roll angle in rad</field>
       <field type="float" name="pitch" units="rad">Pitch angle in rad</field>
       <field type="float" name="yaw" units="rad">Yaw angle in rad</field>
+      <extensions/>
+      <field type="float[21]" name="covariance">Pose covariance matrix upper right triangular (first six entries are the first ROW, next five entries are the second ROW, etc.)</field>
```
Comment (mrivi) [inline message_definitions/v1.0/common.xml]: Could we add also the attitude covariance?

**Comment by mrivi** [comment]
@LorenzMeier  do you agree on bringing this changes in?

File: message_definitions/v1.0/common.xml (hunk 1 above)
Comment (TSC21) [reply message_definitions/v1.0/common.xml]: The attitude quaternion? Cause the pose covariance already includes the RPY covariance.

Comment (mrivi) [reply message_definitions/v1.0/common.xml]: yeah, sorry. I didn't read carefully enough.

**Comment by mrivi** [comment]
@LorenzMeier could you please comment?

**Comment by TSC21** [comment]
Thanks @julianoes!

**Pull request merged**